### PR TITLE
[IE][VPU] Pad layer: enabling of 3D support

### PR DESCRIPTION
**Pull request by rvyunov**
### Details:
 - Enabling of 3D support for Pad layer
 - Added tests to new test infrastructure

### Tickets:
 - 46417

### CODE REVIEW

**Comment by rvyunov** [comment]
@elatkin , @KorA6 , could you please review it?

**Review by elatkin — APPROVED** [review]

File: inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp [hunk 1]
```diff
@@ -93,20 +93,58 @@ void FrontEnd::parsePad(const Model& model, const ie::CNNLayerPtr& _layer, const
     auto layer = std::dynamic_pointer_cast<ie::PadLayer>(_layer);
     IE_ASSERT(layer != nullptr);
 
-    IE_ASSERT(layer->pads_begin.size() == 4);
-    IE_ASSERT(layer->pads_end.size() == 4);
+    const auto ndims = inputs[0]->desc().dimsOrder().numDims();
+    VPU_THROW_UNLESS(ndims == 3 || ndims == 4, "Layer %s support only 3D and 4D input", layer->name);
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp:97]: When you will rebase your branch, please, mention the actual ndims which is provided

File: inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp [hunk 2]
```diff
@@ -93,20 +93,58 @@ void FrontEnd::parsePad(const Model& model, const ie::CNNLayerPtr& _layer, const
...
+
+    IE_ASSERT(layer->pads_begin.size() <= 4);
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp:99]: Please, replace it with more preferable `VPU_THROW_UNLESS` with a descriptive error message

File: inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp [hunk 3]
```diff
@@ -93,20 +93,58 @@ void FrontEnd::parsePad(const Model& model, const ie::CNNLayerPtr& _layer, const
...
+    unsigned int pads_beginH = 0;
+    unsigned int pads_beginW = 0;
+
+    unsigned int pads_endN = 0;
+    unsigned int pads_endC = 0;
+    unsigned int pads_endH = 0;
+    unsigned int pads_endW = 0;
+
+    switch (ndims) {
+        case 4:
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp:113]: It is ok, but I think we can come up with a more elegant solution :)

File: inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp (hunk 1 above)
Comment (rvyunov) [reply inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp]: fixed

File: inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp (hunk 2 above)
Comment (rvyunov) [reply inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp]: fixed

File: inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp (hunk 3 above)
Comment (rvyunov) [reply inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp]: I prepared more elegant solution, please review one more time

**Review by KorA6 — APPROVED** [review]

**Comment by Maxim-Doronin** [comment]
We've just merged another fw. Please, re-build fw from `develop` branch and resolve conflicts. Sorry for inconvenience.

File: inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp (hunk 3 above)
Comment (Maxim-Doronin) [reply inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp]: Thanks, looks great!

**Comment by rvyunov** [comment]
@Maxim-Doronin , there are no regressions, so could you merge it, please?

**Review by Maxim-Doronin — APPROVED** [review]
Great!

**Pull request merged**